app/fissile.go

```diff
@@ -791,77 +791,95 @@ func (f *Fissile) generateAuth(settings kube.ExportSettings) error {
+			continue
+		}
+
+		nodes, err := kube.NewRBACAccount(accountName, settings.RoleManifest.Configuration, settings)
+		if err != nil {
+			return err
+		}
+		err = f.writeHelmNode(authDir, fmt.Sprintf("account-%s.yaml", accountName), nodes...)
+		if err != nil {
+			return err
 		}
 	}
 
+	// Generate roles
 	for roleName, roleSpec := range settings.RoleManifest.Configuration.Authorization.Roles {
-		// Ignore roles referenced by a single instance
-		// group. These are not written as their own files,
-		// but as part of the instance group.
-		if settings.RoleManifest.Configuration.Authorization.RoleUse[roleName] < 2 {
+		var accountNames []string
+		for accountName := range settings.RoleManifest.Configuration.Authorization.RoleUsedBy[roleName] {
+			accountNames = append(accountNames, fmt.Sprintf("- %s", accountName))
+		}
+		if len(accountNames) < 1 {
+			panic(fmt.Sprintf("Role %s used by no accounts", roleName))
+		}
+		if len(accountNames) < 2 {
+			// Ignore roles referenced by a single account. These are not written as their own files,
+			// but as part of the account.
 			continue
 		}
+		sort.Strings(accountNames)
 
-		node, err := kube.NewRBACRole(roleName, roleSpec, settings)
+		node, err := kube.NewRBACRole(roleName, kube.RBACRoleKindRole, roleSpec, settings)
 		if err != nil {
 			return err
 		}
+		node.Set(helm.Comment(fmt.Sprintf(`Role %s used by accounts:\n%s`, roleName, strings.Join(accountNames, "\n"))))
 		err = f.writeHelmNode(authDir, fmt.Sprintf("auth-role-%s.yaml", roleName), node)
 		if err != nil {
 			return err
 		}
 	}
-	for accountName, accountSpec := range settings.RoleManifest.Configuration.Authorization.Accounts {
-		// Ignore accounts referenced by a single instance
-		// group. These are not written as their own files,
-		// but as part of the instance group.
-		if accountSpec.NumGroups < 2 {
+
+	// Generate cluster roles
+	for roleName, roleSpec := range settings.RoleManifest.Configuration.Authorization.ClusterRoles {
+		var accountNames []string
+		for accountName := range settings.RoleManifest.Configuration.Authorization.ClusterRoleUsedBy[roleName] {
+			accountNames = append(accountNames, accountName)
+		}
+		if len(accountNames) < 1 {
+			panic(fmt.Sprintf("Cluster role %s used by no accounts", roleName))
+		}
+		if len(accountNames) < 2 {
+			// Ignore cluster roles referenced by a single account. These are not written as their own files,
+			// but as part of the account.
 			continue
 		}
+		sort.Strings(accountNames)
 
-		nodes, err := kube.NewRBACAccount(accountName, accountSpec, settings)
+		node, err := kube.NewRBACRole(roleName, kube.RBACRoleKindClusterRole, roleSpec, settings)
 		if err != nil {
 			return err
 		}
-		outputPath := filepath.Join(authDir, fmt.Sprintf("account-%s.yaml", accountName))
-		outputFile, err := os.Create(outputPath)
+		node.Set(helm.Comment(fmt.Sprintf(`Cluster role %s used by accounts %s`, roleName, strings.Join(accountNames, "\n"))))
```

Comment:
This comes out as

```
# Cluster role privileged used by accounts default-privileged
# garden-runc
# privileged
```

I would like it better to look like

```
# Cluster role "privileged" used by accounts
# - default-privileged
# - garden-runc
# - privileged
```

I.e.: Quotes around CR name, and the list of accounts looking more like a list, with "-" prefix, and all accounts on their own line.

See also line 832 for the same for roles and accounts.

This would also apparently be more consistent with what I see for service accounts, although that uses two spaces indentation, and should also have quotes around the SA name.

```
# Service account privileged is used by the following instance groups:
#  - nfs-broker
#  - router
```

Generally all the header comments do not seem to use (double)quotes to separate object names from the text around them. IMHO it would be more readable with (double)quotes.

Reply:
Fixed in 12d0ca53.